classes/table/reengagement_participants.php

```diff
@@ -290,33 +290,39 @@ public function query_db($pagesize, $useinitialsbar = true) {
         list($twhere, $tparams) = $this->get_sql_where();
         $psearch = new \mod_reengagement\table\reengagement_search($this->course, $this->context, $this->filterset);
 
-        $total = $psearch->get_total_participants_count($twhere, $tparams);
-
-        $this->pagesize($pagesize, $total);
-
         $sort = $this->get_sql_sort();
 
-        $rawdata = $psearch->get_participants($twhere, $tparams, $sort, $this->get_page_start(), $this->get_page_size());
+        $rs = $psearch->get_participants(
+            $twhere,
+            $tparams,
+            $sort,
+            (int)$this->get_page_start(),
+            (int)$this->get_page_size()
+        );
 
         $this->rawdata = [];
-        foreach ($rawdata as $user) {
+        $total = 0;
+
+        foreach ($rs as $user) {
+            if ($total === 0) {
+                $total = (int)$user->fullcount; // <- confirmed by your dump
```

Comment:
the phpdoc comment here doesn't comply with the coding guidelines - must start with a capital letter and end with a fullstop so will throw codechecker errors - I'd just remove the comment.